### PR TITLE
feat(rpc): implement methods needed for explorer ui

### PR DESCRIPTION
**Pull request by lidatong**

### CODE REVIEW

File: src/ballet/http/fd_http_server.c [hunk 1]
```diff
@@ -649,9 +652,13 @@ write_conn_http( fd_http_server_t * http,
             ulong encoded_len = fd_base64_encode( sec_websocket_accept_base64, sec_websocket_accept, 20 );
             FD_TEST( fd_cstr_printf_check( header_buf, sizeof( header_buf ), &response_len, "HTTP/1.1 101 Switching Protocols\r\nUpgrade: websocket\r\nConnection: Upgrade\r\nSec-WebSocket-Accept: %.*s\r\n\r\n", (int)encoded_len, sec_websocket_accept_base64 ) );
           } else {
-            FD_TEST( fd_cstr_printf_check( header_buf, sizeof( header_buf ), &response_len, "HTTP/1.1 200 OK\r\nContent-Length: %lu\r\nContent-Type: %s\r\n\r\n", conn->response.body_len, conn->response.content_type ) );
+            FD_TEST( fd_cstr_printf_check( header_buf, sizeof( header_buf ), &response_len, "HTTP/1.1 200 OK\r\nContent-Length: %lu\r\nContent-Type: %s\r\nAccess-Control-Allow-Origin: *\r\n\r\n", conn->response.body_len, conn->response.content_type ) );
```
Comment (mmcgee-jump) [inline src/ballet/http/fd_http_server.c:655]: I think this is a bit too specific to the RPC use case. Could you refactor so these response headers are provided in the `fd_http_server_response_t` ?

File: src/ballet/http/fd_http_server.c [hunk 2]
```diff
@@ -12,7 +12,7 @@
 #include <sys/socket.h>
 #include <netinet/in.h>
 
-// #define FD_HTTP_SERVER_DEBUG 1
+#define FD_HTTP_SERVER_DEBUG 1
```
Comment (mmcgee-jump) [inline src/ballet/http/fd_http_server.c:15]: dont check in

**Review by mmcgee-jump — CHANGES_REQUESTED** [review]
.

File: src/ballet/http/fd_http_server.c (hunk 1 above)
Comment (lidatong) [reply src/ballet/http/fd_http_server.c:655]: will clean up once ready

File: src/app/rpcserver/fd_rpc_service.c [hunk 3]
```diff
@@ -1323,10 +1468,21 @@ method_getTransaction(struct json_values* values, fd_rpc_ctx_t * ctx) {
 // Implementation of the "getTransactionCount" methods
 static int
 method_getTransactionCount(struct json_values* values, fd_rpc_ctx_t * ctx) {
-  (void)values;
-  (void)ctx;
-  fd_webserver_t * ws = &ctx->global->ws;
-  fd_web_error(ws, "getTransactionCount is not implemented");
+  FD_METHOD_SCRATCH_BEGIN( 1<<28 ) { /* read_epoch consumes a ton of scratch space! */
```
Comment (asiegel-jt) [inline src/app/rpcserver/fd_rpc_service.c:1471]: Use the transaction count from last_notify

File: src/app/rpcserver/fd_rpc_service.c [hunk 4]
```diff
@@ -2158,6 +2323,39 @@ fd_rpc_replay_notify(fd_rpc_ctx_t * ctx, fd_replay_notif_msg_t * msg) {
 
   if( msg->type == FD_REPLAY_SLOT_TYPE ) {
     fd_readwrite_start_write( &subs->lock );
+
+    long ts = fd_log_wallclock() / (long)1e9;
+    if( FD_UNLIKELY( ts - subs->perf_sample_ts >= 60 ) ) {
```
Comment (asiegel-jt) [inline src/app/rpcserver/fd_rpc_service.c:2328]: loop. you may need multiple pops

File: src/app/rpcserver/main.c [hunk 5]
```diff
@@ -108,6 +108,8 @@ signal1( int sig ) {
 
 int main( int argc, char ** argv ) {
   fd_boot( &argc, &argv );
+#include "../../flamenco/fd_flamenco.h"
```
Comment (asiegel-jt) [inline src/app/rpcserver/main.c:111]: remove

File: src/ballet/http/fd_http_server.c [hunk 6]
```diff
@@ -649,9 +652,33 @@ write_conn_http( fd_http_server_t * http,
...
+          FD_TEST( fd_cstr_printf_check( header_buf,
+                                         sizeof( header_buf ),
+                                         &response_len,
+                                         "HTTP/1.1 204 No Content\r\n"
+                                         "Access-Control-Allow-Origin: %s\r\n"
+                                         "Access-Control-Allow-Methods: %s\r\n"
+                                         "Access-Control-Allow-Headers: %s\r\n"
+                                         "Access-Control-Max-Age: %lu\r\n"
+                                         "\r\n",
+                                         conn->response.access_control_allow_origin,
```
Comment (asiegel-jt) [inline src/ballet/http/fd_http_server.c:676]: check for nulls

File: src/app/rpcserver/fd_rpc_service.c (hunk 3 above)
Comment (lidatong) [reply src/app/rpcserver/fd_rpc_service.c:1471]: doesnt work because user passes in commitment level

File: src/ballet/http/fd_http_server.c (hunk 1 above)
Comment (lidatong) [reply src/ballet/http/fd_http_server.c]: @mmcgee-jump take another look

**Review by lheeger-jump — APPROVED** [review]

**Review by mmcgee-jump — APPROVED** [review]

**Pull request merged**